app/components/domains-table.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  Table,
+  Tr,
+  Th,
+  Thead,
+  Tbody,
+  TableContainer,
+  Td,
+  IconButton,
+  Flex,
+  Tooltip,
+  Card,
+  useToast,
+} from '@chakra-ui/react';
+import type { Record, RecordStatus } from '@prisma/client';
+import {
+  EditIcon,
+  DeleteIcon,
+  CheckCircleIcon,
+  RepeatIcon,
+  CopyIcon,
+  TimeIcon,
+  WarningIcon,
+} from '@chakra-ui/icons';
+
+interface DomainsTableProps {
+  domains: Record[];
+  onAction: (domaindID: number, action: DomainsTableAction) => void;
+}
+
+export type DomainsTableAction = 'EDIT' | 'DELETE' | 'RENEW';
+
+export default function DomainsTable(props: DomainsTableProps) {
+  const { domains, onAction } = props;
+
+  const toast = useToast();
+
+  function onCopyNameToClipboard(name: string) {
+    navigator.clipboard.writeText(name);
+    toast({
+      title: 'Name was copied to clipboard',
+      position: 'bottom-right',
+      status: 'success',
+    });
+  }
+
+  function renderDomainStatus(action: RecordStatus) {
+    switch (action) {
+      case 'active':
+        return (
+          <Tooltip label="Domain is live">
+            <CheckCircleIcon color="green.500" boxSize="6" />
+          </Tooltip>
+        );
+      case 'error':
+        return (
+          <Tooltip label="Domain error">
+            <WarningIcon color="brand.500" boxSize="6" />
+          </Tooltip>
+        );
+      case 'pending':
+        return (
+          <Tooltip label="Domain is pending">
+            <Flex
+              width="6"
+              height="6"
+              borderRadius="20"
+              alignItems="center"
+              justifyContent="center"
+              background="yellow.500"
+            >
+              <TimeIcon color="white" boxSize="3" />
+            </Flex>
+          </Tooltip>
+        );
+    }
+  }
+
+  return (
+    <Card p="2" mt="4">
+      <TableContainer>
+        <Table variant="striped" colorScheme="gray">
+          <Thead>
+            <Tr>
+              <Th />
+              <Th>Name</Th>
+              <Th>Type</Th>
+              <Th>Value</Th>
+              <Th>Expiration date</Th>
+              <Th />
+            </Tr>
+          </Thead>
+          <Tbody>
+            {domains.map((domain) => {
+              return (
```

Comment:
Nit:
```suggestion
            {domains.map((domain) => (
```

...you need to update the end too to match, this is just a tiny bit cleaner